afqinsight/match.py

```diff
@@ -0,0 +1,139 @@
+"""Perform statistical matching comparing test and control units."""
+
+from os import name
+import numpy as np
+import pandas as pd
+from scipy.optimize import linear_sum_assignment
+
+
+__all__ = ["mahalonobis_dist_match"]
+
+
+def mahalonobis_dist_match(
+    data=None, test=None, ctrl=None, status_col=None, feature_cols=None, threshold=0.2
+):
+    """
+    If data is None, then test and ctrl must be arrays.
+    If data is provided, then status_col and feature_cols are used.
+
+    Parameters
+    ----------
+    data : pandas Dataframe, optional
+        The feature samples for test subjects.
+        Default: None
+    test : array-like of shape (n_samples1, n_features), optional
+        The feature samples for test subjects.
+        Default: None
+    ctrl : array-like of shape (n_samples2, n_features), optional
+        Feature samples for control subjects from which
+        matches will be selected.
+        Default: None
+    status_col : str, optional
+        Column with subject status. Must have only two unique values,
+        evaluating to True and False, where True indicates a test subject
+        and False indicates a control.
```

Comment:
See suggestion below allowing non boolean values.
```suggestion
        Column with subject status. Must have only two unique values.
```